### PR TITLE
pkp/pkp-lib#8406 Improve DublinCoreMetaPlugin

### PR DESCRIPTION
**Pull request by bozana**
s. https://github.com/pkp/pkp-lib/issues/8406

### CODE REVIEW

File: plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php [hunk 1]
```diff
@@ -72,85 +71,94 @@ public function articleView($hookName, $args)
...
+                if (!empty($coverage)) {
+                    $templateMgr->addHeader('dublinCoreCoverage' . $locale, '<meta name="DC.Coverage" xml:lang="' . htmlspecialchars(substr($locale, 0, 2)) . '" content="' . htmlspecialchars(strip_tags($coverage)) . '"/>');
+                }
             }
         }
 
-        $i = 0;
-        foreach ($publication->getData('authors') as $author) {
-            $templateMgr->addHeader('dublinCoreAuthor' . $i++, '<meta name="DC.Creator.PersonalName" content="' . htmlspecialchars($author->getFullName(false)) . '"/>');
+        if ($authors = $publication->getData('authors')) {
```
Comment (asmecher) [inline plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php:97]: Is this extra `if` necessary? `$publication->getData('authors')` should always be a collection; if it's empty, that's fine, no authors get added to the DC metadata.

File: plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php [hunk 2]
```diff
@@ -72,85 +71,94 @@ public function articleView($hookName, $args)
...
+                }
             }
         }
 
-        $i = 0;
-        if ($coverages = $article->getCoverage(null)) {
+        if ($coverages = $publication->getData('coverage')) {
             foreach ($coverages as $locale => $coverage) {
-                $templateMgr->addHeader('dublinCoreCoverage' . $i++, '<meta name="DC.Coverage" xml:lang="' . htmlspecialchars(substr($locale, 0, 2)) . '" content="' . htmlspecialchars(strip_tags($coverage)) . '"/>');
+                if (!empty($coverage)) {
```
Comment (asmecher) [inline plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php:91]: This probably isn't a case where it'll get us, but watch out for `empty` and values like `"0"` (string zero) or `0` (numeric zero). `empty('0')` and `empty(0)` both return `true`. Better to use `if (coverage != '')` (this is safe with nulls and numeric zeros too). (Also elsewhere)

File: plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php [hunk 3]
```diff
@@ -72,85 +71,94 @@ public function articleView($hookName, $args)
...
+        if ($dateModified = $publication->getData('lastModified')) {
             $templateMgr->addHeader('dublinCoreDateModified', '<meta name="DC.Date.modified" scheme="ISO8601" content="' . date('Y-m-d', strtotime($dateModified)) . '"/>');
         }
 
-        if ($publication) {
-            $i = 0;
-            if ($abstracts = (array) $publication->getData('abstract')) {
-                foreach ($abstracts as $locale => $abstract) {
-                    $templateMgr->addHeader('dublinCoreAbstract' . $i++, '<meta name="DC.Description" xml:lang="' . htmlspecialchars(substr($locale, 0, 2)) . '" content="' . htmlspecialchars(strip_tags($abstract)) . '"/>');
+        if ($abstracts = $publication->getData('abstract')) {
```
Comment (asmecher) [inline plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php:114]: As above, is the `if` necessary? (Also elsewhere.)

File: plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php [hunk 4]
```diff
@@ -159,39 +167,43 @@ public function articleView($hookName, $args)
...
-                    continue;
+        if ($keywords = $publication->getData('keywords')) {
+            foreach ($keywords as $locale => $localeKeywords) {
+                foreach ($localeKeywords as $i => $keyword) {
+                    $templateMgr->addHeader('dublinCoreKeyword' . $locale . $i++, '<meta name="DC.Subject" xml:lang="' . htmlspecialchars(substr($locale, 0, 2)) . '" content="' . htmlspecialchars($keyword) . '"/>');
                 }
+            }
+        }
+
+        $templateMgr->addHeader('dublinCoreTitle', '<meta name="DC.Title" content="' . htmlspecialchars($publication->getLocalizedFullTitle($publicationLocale)) . '"/>');
```
Comment (asmecher) [inline plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php:188]: Watch out for a collision with @touhidurabir's work on https://github.com/pkp/pkp-lib/issues/2564 here!

**Review by asmecher — CHANGES_REQUESTED** [review]
Looks good! Just a few comments.

**Comment by asmecher** [comment]
I've reviewed this, but note that I haven't tested it! I see some changes around e.g how keywords are handled etc.; I assume those are working OK on your system, but logically they make sense to me.

File: plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php (hunk 1 above)
Comment (bozana) [reply plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php:97]: The metadata that can be enabled and disabled, like subjects, keywords and type, can be null, so for them I left this `if`.

**Pull request merged**